src/olympia/blocklist/mlbf.py

```diff
@@ -245,17 +246,52 @@ def generate_diffs(
         }
 
     def generate_and_write_stash(self, previous_mlbf: 'MLBF' = None):
-        # compare previous with current blocks
-        extras, deletes, _ = self.generate_diffs(previous_mlbf)[BlockType.BLOCKED]
+        """
+        Generate and write the stash file representing
+        changes between the previous and current MLBF Filters.
+
+        In order to support FX clients that don't support soft blocking,
+        unblocked is a union of deletions from blocked and deletions from soft_blocked
+        filtering out any items that are in the newly blocked list.
+
+        Versions that move from hard to soft blocked will be picked up by old clients
+        as no longer hard blocked by being in the unblocked list.
+
+        bug: https://bugzilla.mozilla.org/show_bug.cgi?id=soft-blocking
+        Clients supporting soft blocking will also see soft blocked items as unblocked
+        but then will check the soft blocked list and re-add them to soft blocked.
+
+        Versions that move from soft to hard blocked will be picked up by
+        all clients in the blocked list. Note, even though the version is removed
+        from the soft blocked list, it is important that we do not include it
+        in the "unblocked" stash (like for hard blocked items) as this would
+        result in the version being in both blocked and unblocked stashes.
+        """
+        diffs = self.generate_diffs(previous_mlbf)
+        blocked_added, blocked_removed, _ = diffs[BlockType.BLOCKED]
         stash_json = {
-            'blocked': extras,
-            'unblocked': deletes,
+            'blocked': blocked_added,
+            'unblocked': blocked_removed,
         }
+
+        if waffle.switch_is_active('mlbf-soft-blocks-enabled'):
```

Comment:
Ha, I see. So there is the waffle switch. Given that the migration file is short, I would just add it in this patch, rather than in its own PR without anything else.

Reply:
It's referenced in both PRs. Why not just land it?